Softeq.XToolkit.PushNotifications.iOS/Services/IosPushNotificationService.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Developed by Softeq Development Corporation
+// http://www.softeq.com
+
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Foundation;
+using Softeq.XToolkit.Common.Logger;
+using Softeq.XToolkit.PushNotifications.Abstract;
+using Softeq.XToolkit.PushNotifications.iOS.Abstract;
+using UIKit;
+using UserNotifications;
+
+namespace Softeq.XToolkit.PushNotifications.iOS.Services
+{
+    /// <summary>
+    ///     Default implementation of <see cref="IPushNotificationsService"/> interface for iOS platform.
+    /// </summary>
+    public sealed class IosPushNotificationService : IPushNotificationsService, IPushNotificationAppDelegate
+    {
+        private readonly IPushNotificationsConsumer _pushNotificationsConsumer;
+        private readonly ILogger _logger;
+
+        /// <summary>
+        ///     Initializes a new instance of the <see cref="IosPushNotificationService"/> class.
+        /// </summary>
+        /// <param name="pushNotificationsConsumer">Consumer of the push notification related callbacks.</param>
+        /// <param name="logManager">Provides logging.</param>
+        public IosPushNotificationService(
+            IPushNotificationsConsumer pushNotificationsConsumer,
+            ILogManager logManager)
+        {
+            _pushNotificationsConsumer = pushNotificationsConsumer;
+            _logger = logManager.GetLogger<IosPushNotificationService>();
+
+            UNUserNotificationCenter.Current.Delegate = new UserNotificationCenterDelegate(pushNotificationsConsumer, logManager);
+            UNUserNotificationCenter.Current.SetNotificationCategories(new NSSet<UNNotificationCategory>(pushNotificationsConsumer.GetCategories().ToArray()));
```

Comment:
Bu the time this code is called you already need to have your consumer ready (composite or otherwise), and it is not supported to change the behavior of the comsumers in runtime.
Also, it is recommended to execute this code as soon as possible - not to miss any notifications